src/sentry/grouping/strategies/newstyle.py

```diff
@@ -899,3 +906,29 @@ def threads_variant_processor(
     variants: ReturnedVariants, context: GroupingContext, **meta: Any
 ) -> ReturnedVariants:
     return remove_non_stacktrace_variants(variants)
+
+
+def react_hydration_error_with_cause(exceptions: list[SingleException]) -> int | None:
+    main_exception_id = None
+    # Starting with React 19, hydration errors can contain a cause error which is useful to display
+    if (
+        exceptions[0].type == "Error"
+        and exceptions[0].value
+        == "There was an error while hydrating but React was able to recover by instead client rendering from the nearest Suspense boundary."
```

Comment:
is it guaranteed to always be just this message and nothing else?

Reply:
At the moment, that's what we have noticed.
We may be fine with dropping checking the `value` but I need data to determine what other platforms could be impacted.

I assume it would be fine but I just don't know.

Reply:
@vartec I have added the other test case based on the data @eps1lon has provided.